picard/ui/coverartbox.py

```diff
@@ -545,6 +550,71 @@ def load_remote_image(self, url, data):
             log.warning("Can't load image: %s", e)
             return
 
+    def delete_cover_art(self):
+        if not self.item or not self.item.metadata.images:
+            if not self.item.orig_metadata.images:
+                return
+
+        cover_art_list = [image.source or _("Unnamed Cover Art") for image in self.item.metadata.images]
+
+        selected_item, ok_pressed = QtWidgets.QInputDialog.getItem(self, _("Delete Cover Art"),
+                                                                    _("Select the cover art image to delete:"),
+                                                                    cover_art_list, 0, False)
+        if ok_pressed:
+            selected_image_index = cover_art_list.index(selected_item)
+            selected_image = self.item.metadata.images[selected_image_index]
+            try:
+                self.delete_cover_art_for_item(self.item, selected_image)
+            except CoverArtImageError as e:
+                log.error("Can't delete image: %s", e)
+                return
+
+    def delete_cover_art_for_item(self, item, selected_image):
+        metadata = item.metadata
+        if not metadata or not metadata.images:
+            return
+
+        debug_info = "Deleting %r from %r"
```

Comment:
Should this be "Deleted %r from %r", since it isn't logged until the deletion is complete?